app/assets/javascripts/discourse-common/lib/helpers.js.es6

```diff
@@ -45,20 +45,25 @@ function resolveParams(ctx, options) {
   return params;
 }
 
-export function registerUnbound(name, fn) {
-  const func = function(property, options) {
-    if (
-      options.types &&
-      (options.types[0] === "ID" || options.types[0] === "PathExpression")
-    ) {
-      property = get(this, property, options);
+export function registerUnbound(name, fn, argCount = 1) {
```

Comment:
I don't think we should lightly change the signature of this function, since it's used in many places.

Is it not possible to get the arity by checking `options.types.length` perhaps?

Reply:
The new `argCount` argument has a default, so it shouldn't have any impact on existing uses. I originally tried using `options.length`, but it didn't work because some existing places register like this:

```
registerUnbound('test', function(value){ ... });
```
while others want to use the keyword arguments, so they register like this:
```
registerUnbound('test', function(value, params){ ... });
```
So number of arguments of the handler is not a good indication of the number of arguments passed to the handlebars helper.

Reply:
Resolved in d7daf54